docs: Add release notes for v1.7.0.

davecgh opened the pull request:
~This is currently the release notes for rc1 for the purposes of review.  It will be updated to remove the RC status and merged once the final non-rc release is ready.~

jzbz commented:
full administrative priveleges. -> privileges.

rstaudt2 commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+
+### Revert Treasury Maximum Expenditure Policy Vote
+
+The first new vote available as of this released has the id `reverttreasurypolicy`.
```
```suggestion
The first new vote available as of this release has the id `reverttreasurypolicy`.
```

Same typo for the other votes as well.

davecgh replied:
Fixed.

rstaudt2 commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+
+The first new vote available as of this released has the id `reverttreasurypolicy`.
+
+The primary goal is this change is to revert the currently active maximum
```
```suggestion
The primary goal of this change is to revert the currently active maximum
```

davecgh replied:
Done.

degeri commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+
+#### Unspent Transaction Output Cache
+
+All reads and write of unspent transaction outputs (utxos) now go through a
```
```suggestion
All reads and writes of unspent transaction outputs (utxos) now go through a
```

degeri replied:
should both be plural or singular..

davecgh replied:
Updated.

degeri commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+
+* For reads, the UTXO cache acts as a read-through cache.  All UTXO reads go
+  through the cache.  When there is a cache miss, the cache loads the missing
+  data from disk and caches it for future lookups.
```
```suggestion
  data from the disk and caches it for future lookups.
```

davecgh replied:
Added, although I'm fairly certain the article is optional here.

degeri commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+  data from disk and caches it for future lookups.
+* For writes, the UTXO cache acts as a write-back cache.  Writes to the cache
+  are acknowledged by the cache immediately, but are only periodically flushed
+  to disk.
```
```suggestion
  to the disk.
```

davecgh replied:
Updated, but same as before.

degeri commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+
+* An additional `tree` parameter is now required in order to explicitly identify
+  the exact transaction output being requested
+* The transaction `version` field is no longer available in primary JSON object
```
```suggestion
* The transaction `version` field is no longer available in the primary JSON object
```

davecgh replied:
Added.

degeri commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+
+### Reject Protocol Message Deprecated (`reject`)
+
+The `reject` peer-to-peer protocol message is now deprecated and are scheduled
```
```suggestion
The `reject` peer-to-peer protocol message is now deprecated and is scheduled
```

davecgh replied:
Fixed.

jholdstock commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+Four new consensus change votes are now available as of this release.  After
+upgrading, stakeholders may set their preferences through their wallet.
+
+### Revert Treasury Maximum Expenditure Policy Vote
```
Suggest making all four of these headings at fourth level (`####`) so they are nested under the heading `### Four New Consensus Change Votes`

davecgh replied:
Changed.  I went back and forth on this and had decided to leave it with 3 so they stand out better as 4 makes it a bit small, but I agree that they really should be sub headings.

jholdstock commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+  * A more strict cross origin request policy
+  * A new alternative client authentication mechanism based on TLS certificates
+  * Availability of the scripting language version for transaction outputs
+  * Various other RPC modifications, additions, and removals
```
Suggest removing this line. 

```
Some of the key highlights are:

...

- Various other RPC modifications, additions, and removals
```

"Various modifications" is not a highlight

davecgh replied:
Hmm, I'm not sure I agree here.  There are several modifications to the results of various RPCs that RPC clients need to be aware of, hence this is highlighting that so they need to look further into them.  Calling each one out individually in the highlights doesn't seem appropriate given the nature of it being a short summary.

Do you have any suggestions for a better way to word that in a succinct manner?

jholdstock replied:
Fair point. Leave it as-is then, or maybe go with `Various other *important* RPC modifications, additions, and removals`?

davecgh replied:
After thinking on it a bit, how about just changing the `RPC` to `API` since it's already under a bullet point regarding updates to the RPC server?

jholdstock replied:
Yeah good idea.

davecgh replied:
Updated this to:

> Several other notable updates, additions, and removals related to the JSON-RPC API

jholdstock commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+* Provide an easy, reliable, and efficient method for software and hardware to
+  determine exactly which rules should be applied to transaction and script
+  versions
+* Further embrace the increased security, and other desirable properties, that
```
I don't think this sentence needs any commas.

davecgh replied:
Updated.

jholdstock commented on docs/release-notes/release-notes-1.7.0.md:
```diff
@@ -0,0 +1,1434 @@
+The fourth new vote available as of this release has the id `changesubsidysplit`.
+
+The proposed modification to the subsidy split is intended to substantially
+diminish the ability to attack Decred's markets with mined coins and improve
```
Mentioning markets here doesn't sit right with me. This is a software release document and IMO should be focused on the technical aspects of the release - changes to the software, new features etc.

I'd keep this section focused on the nature of the proposed consensus change and not mention market/attack theories. There is a link to the proposal just below this for those who are interested in reading further justification for the change.

davecgh replied:
This section serves to quickly highlight the motivation for changes at a quick glance as opposed to technically describe them.  Take a look at all of the other consensus changes in this section and they are all "The primary goal(s) of this change...." and/or "The intention of this ...".

This description is directly from the motivation section of the DCP, so I don't see any reason to avoid mentioning it here.

rstaudt2 reviewed (approved):
Looks great to me.  It must have taken quite some time to go through all those PRs.

Since there are so many performance enhancements in this release, I ran some initial sync tests to see the overall relative difference in performance between `release-v1.6` and `release-v1.7`.  Based on my testing:

- `release-v1.7` is **~47.87% faster** than `release-v1.6` with default settings
- `release-v1.7` is **~39.36% faster** than `release-v1.6` when using the latest checkpoint from 1.6 (this is to compare without giving `release-v1.7` the advantage of a more recent checkpoint)
- `release-v1.7` is **~29.37% faster** than `release-v1.6` with checkpoints disabled

Details of the tests I ran are in this [gist](https://gist.github.com/rstaudt2/e716f09eb9193fff6d8397f181ec3320).

dnldd reviewed (approved):
Reads well. Only thing I noticed which could be improved is punctuation for list items with sentences, a few are punctuated while majority are not.

davecgh commented:
Thanks for those numbers @rstaudt2.  I added a section entitled `Substantially Reduced Initial Sync Time` that uses the first value since that is what most users reading the release notes can expect to observe.  It's nice to see the other numbers are well.  The difference between checkpoints and no checkpoints shows pretty clearly that utxos are what dominates.

I went back and forth a bit on whether to include the others, but I think it's probably better to keep it more high level to strike a balance.

---

I also updated the `Unspent Transaction Output Cache` section to make the bullet points more succinct and  consistent with the rest (single sentences, no periods) per @dnldd.

rstaudt2 commented:
> I added a section entitled `Substantially Reduced Initial Sync Time` that uses the first value since that is what most users reading the release notes can expect to observe.

Nice, that looks good to me.  Agreed that the first value makes the most sense to put there.
